business/istio_status.go

```diff
@@ -223,3 +210,62 @@ func GetDeploymentStatus(d apps_v1.Deployment) string {
 	}
 	return status
 }
+
+func (iss *IstioStatusService) getAddonComponentStatus() IstioComponentStatus {
+	var wg sync.WaitGroup
+	wg.Add(4)
+
+	staChan := make(chan IstioComponentStatus, 4)
+	extServices := config.Get().ExternalServices
+	ics := IstioComponentStatus{}
+
+	go getAddonStatus("prometheus", true, extServices.Prometheus.URL, &extServices.Prometheus.Auth, true, staChan, &wg)
+	go getAddonStatus("grafana", extServices.Grafana.Enabled, extServices.Grafana.InClusterURL, &extServices.Grafana.Auth, extServices.Grafana.IsCoreComponent, staChan, &wg)
+	go getAddonStatus("jaeger", extServices.Tracing.Enabled, extServices.Tracing.InClusterURL, &extServices.Tracing.Auth, extServices.Tracing.IsCoreComponent, staChan, &wg)
+
+	// Custom dashboards may use the main Prometheus config
+	customProm := extServices.CustomDashboards.Prometheus
+	if customProm.URL == "" {
+		customProm = extServices.Prometheus
+	}
+	go getAddonStatus("custom dashboards", extServices.CustomDashboards.Enabled, customProm.URL, &extServices.CustomDashboards.Prometheus.Auth, extServices.CustomDashboards.IsCoreComponent, staChan, &wg)
```

Comment:
It's not `&extServices.CustomDashboards.Prometheus.Auth` but `&customProm.Auth` then :)